packages/frontend/__tests__/components/forms/signup-form.test.js

```diff
@@ -26,24 +26,52 @@ describe('Signup Form Tests', () => {
     const { getByText, getByLabelText } = render(
       <SignupForm onSubmit={onSubmit} />
     );
-    const nameInput = getByLabelText('Name');
-    const email = getByLabelText('Email');
-    const password = getByLabelText('Password');
-    const passwordConfirmation = getByLabelText('Password Confirmation');
+
     const button = getByText('Create Account');
 
-    await userEvent.type(nameInput, 'Harry Potter');
-    await userEvent.type(email, 'harry@potter.com');
-    await userEvent.type(password, '#49sd2YXBKX%XZ');
-    await userEvent.type(passwordConfirmation, '#49sd2YXBKX%XZ');
+    const inputs = [
+      { item: getByLabelText('Name'), value: 'Harry Potter' },
+      { item: getByLabelText('Email'), value: 'harry@potter.com' },
+      { item: getByLabelText('Password'), value: '123456' },
+      { item: getByLabelText('Password Confirmation'), value: '123456' },
+    ];
+
+    inputs.forEach(async input => {
+      await userEvent.type(input.item, input.value);
+    });
+
     await wait();
 
     expect(button).not.toHaveAttribute('disabled');
-    await userEvent.click(button);
+    userEvent.click(button);
     await wait();
     expect(onSubmit).toHaveBeenCalledTimes(1);
   });
 
+  test('Create account button shouldnt call func if onSubmit didnt defined', async () => {
+    const onSubmit = jest.fn();
+    const { getByText, getByLabelText } = render(<SignupForm />);
+
+    const button = getByText('Create Account');
+
+    const inputs = [
+      { item: getByLabelText('Name'), value: 'Harry Potter' },
+      { item: getByLabelText('Email'), value: 'harry@potter.com' },
+      { item: getByLabelText('Password'), value: '123456' },
+      { item: getByLabelText('Password Confirmation'), value: '123456' },
+    ];
+
+    inputs.forEach(async input => {
+      await userEvent.type(input.item, input.value);
+    });
```

Comment:
I would prefer repeating the same code rather than a smarter solution in a test to make it more readable but it's not a show stopper to me.